File: uva_471.py
MAX=9876543210
def check(n):
    digit=[0 for _ in range(10)]
    while n:
        if not digit[n%10]:
            digit[n%10]=True
        else:
            return False
        n=int(n/10)
    return True
num=int(input())
while num:
    num-=1
    temp=input()
    N=int(input())
    S2=1
    while S2<MAX:
        S1=S2*N
        if check(S1) and check(S2):
            print('{} / {} = {}'.format(S1, S2, N))
        if S1>MAX:
            break
        S2+=1
    if num:
        print('')

File: uva_471.cpp
# include <iostream>
# define MAX 9876543210
using namespace std;
bool check(long long int n){
    int digit[10]={0};
    while(n){
        if (!digit[n%10])
            digit[n%10]=true;
        else
            return false;
        n/=10;
    }
    return true;
}
int main(void){
    int num;
    cin >> num;
    while(num--){
        long long int S1=0, S2, N;
        cin >> N;
        for (S2=1; S2<MAX; ++S2){
            S1=S2*N;
            if (check(S1) && check(S2))
                cout << S1<<" / "<<S2<<" = "<<N<<endl;
            if (S1>MAX)
                break;
        }
        if (num)
            cout << endl;
    }
    return 0;
}